src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,182 @@
+package storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import exception.DukeException;
+import parser.DateTimeParser;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.TaskList;
+import task.Todo;
+
+/**
+ * <h1>Storage class</h1>
+ * Stores and reads tasks generated by the user to ensure the
+ * tasks can be used the next time the chat bot is run.
+ */
+public class Storage {
+    private String folderPath;
+    private String filename;
+    private String userDirectory = System.getProperty("user.dir");
+
+    /**
+     * Creates a Storage object.
+     *
+     * @param folderPath folder path the output is to be saved at.
+     * @param filename file name the output is to be saved as.
+     */
+    public Storage(String folderPath, String filename) {
+        this.folderPath = folderPath;
+        this.filename = filename;
+    }
+
+    /**
+     * Returns an ArrayList of Tasks containing the saved Tasks, if they exist.
+     *
+     * @return ArrayList of Tasks
+     * @throws DukeException if an error is encountered when the file
+     *         is unable to be read.
+     */
+    public ArrayList<Task> readSavedTasks() throws DukeException {
+        Path dataDirectoryPath = Paths.get(userDirectory, folderPath);
+        Path savedTasksPath = Paths.get(userDirectory, folderPath, filename);
+        if (!hasDataDirectory(dataDirectoryPath)) {
+            createDataDirectory(dataDirectoryPath);
+            return new ArrayList<>();
+        }
+        if (!hasSavedTasks(savedTasksPath)) {
+            return new ArrayList<>();
+        } else {
+            File tasksFile = getTasksFile(savedTasksPath);
+            ArrayList<Task> tasks = new ArrayList<>();
+            try {
+                tasks = parseFileToTasks(tasksFile);
+            } catch (FileNotFoundException e) {
+                throw new DukeException(e.getMessage());
+            } finally {
+                return tasks;
+            }
+        }
+    }
+
+    private File getTasksFile(Path savedTasksPath) {
+        File tasksFile = new File(getAbsolutePath(savedTasksPath));
+        return tasksFile;
+    }
+
+    private ArrayList<Task> parseFileToTasks(File tasksFile) throws FileNotFoundException, DukeException {
+        Scanner sc = new Scanner(tasksFile);
+        ArrayList<Task> tasks = new ArrayList<>();
+        while (sc.hasNext()) {
+            String taskDescription = sc.nextLine();
+            tasks.add(parseLineToTask(taskDescription));
+        }
+        return tasks;
+    }
+
+    private Task parseLineToTask(String line) throws DukeException {
+        if (line.length() <= 7) {
```

Comment:
This value of 7 seems out of place. Would it be better to reference it to a static value?